tests/integration/Repositories/DuplicateTest.php

```diff
@@ -0,0 +1,175 @@
+
+        $this->assertCount(2, $module->getModelClassName()::get());
+    }
+
+    public function testDuplicateWithRevisionsAndBrowser(): void
+    {
+        $browserModule = AnonymousModule::make('trees', $this->app)
+            ->withFields(['title'])
+            ->boot();
+
+        $module = AnonymousModule::make('aleaves', $this->app)
+            ->withRevisions()
+            ->withFields(['title' => ['translatable' => true]])
+            ->withBelongsToMany(['trees' => $browserModule->getModelClassName()])
+            ->boot();
+
+        $model = $module->getRepository()->create([
+            'title' => ['en' => 'English title'],
+            'active' => ['en' => true],
+            'browsers' => [
+                'trees' => [
+                    ['id' => $treeId = $browserModule->getModelClassName()::create(['title' => 'demo'])->id],
+                ],
+            ],
+        ]);
+
+        $this->assertCount(1, $model->trees);
+        $this->assertEquals($treeId, $model->trees->first()->id);
+
+        $duplicate = $module->getRepository()->duplicate($model->id);
+
+        $this->assertCount(1, $duplicate->trees);
+        $this->assertEquals($treeId, $duplicate->trees->first()->id);
+    }
+
+    public function testDuplicateWithRevisionsAndBlocksAndJsonRepeaters(): void
+    {
+        $module = AnonymousModule::make('bleaves', $this->app)
+            ->withRevisions()
+            ->withFields(['title' => ['translatable' => true], 'repeaterdata' => ['type' => 'json']])
+            ->boot();
+
+        $model = $module->getRepository()->create([
+            'title' => ['en' => 'English title'],
+            'active' => ['en' => true],
+            'blocks' => [
+                [
+                    'type' => 'a17-block-quote',
+                    'content' => [
+                        'quote' => 'Quote',
+                        'author' => 'Variable the first',
+                    ],
+                    'id' => time(),
+                ],
+            ],
+            'repeaters' => [
+                'repeaterdata' => [
+                    [
+                        'id' => time() + 1,
+                        'content' => [
+                            'description' => 'Hello world!',
+                        ],
+                    ],
+                ],
+            ],
+        ]);
+
+        $this->assertCount(1, $model->repeaterdata);
+        $this->assertEquals('English title', $model->title);
+        $this->assertCount(1, $model->blocks);
+
+        $this->assertCount(1, Block::get());
+
+        $duplicate = $module->getRepository()->duplicate($model->id);
+
+        $this->assertCount(1, $model->repeaterdata);
+        $this->assertCount(1, $duplicate->blocks);
+        $this->assertCount(2, Block::get());
+
+        $this->assertNotEquals($duplicate->blocks->first()->id, $model->blocks->first()->id);
+    }
+    
+    public function testDuplicateWithRevisionsAndRepeaters(): void
```

Comment:
Hey @ifox this one currently fails, but I am not sure it should?

When you duplicate a module that has a repeater, should the new model refer the same repeater, clone the repeater or do nothing with the repeater?

Speaking about current behaviour

Reply:
Current behavior is creating a new record from a cloned submission, so the expectation is that it would clone the repeater rows as well.